pkg/ccl/backupccl/backup_planning.go

```diff
@@ -1051,7 +1051,17 @@ func backupPlanHook(
 			CreatedBy: backupStmt.CreatedByInfo,
 		}
 
+		var sj *jobs.StartableJob
 		if backupStmt.Options.Detached {
+			// The protect timestamp logic for a DETACHED BACKUP can be run within the
+			// same txn as the BACKUP is being planned in, because we do not wait for
+			// the BACKUP job to complete.
+			sj, err = protectTimestampForBackup(ctx, p, jr, p.ExtendedEvalContext().Txn, resultsCh, spans,
```

Comment:
your new helper creates the job, right?
but doesn't `StartAsyncJob` also create a job below? I'm a bit confused

Honestly I'm not sure what `StartAsyncJob` is buying us -- i might just inline it here/in restore instead, so you could do something like

```
var resRow Datums
if err := db.Txn(txn => 
  j := CreateAdoptableJob(txn) 
  protectTS(txn, j.ID)
); err != nil {
return err 
}
resultsCh <- resRow
```